### PR TITLE
Fix machine translation updates to allow any user to edit.

### PR DESCRIPTION
**Pull request by NateMerritt**
See https://biblionexusworkspace.slack.com/archives/C0662HZ7RKM/p1734458335235189

### CODE REVIEW

File: src/Aquifer.API/Endpoints/Resources/Content/MachineTranslation/Update/Endpoint.cs [hunk 1]
```diff
@@ -19,13 +19,12 @@ public override async Task HandleAsync(Request req, CancellationToken ct)
         var existingMt = await dbContext.ResourceContentVersionMachineTranslations.AsTracking()
             .FirstOrDefaultAsync(x =>
                     x.Id == req.Id &&
-                    x.ResourceContentVersion.AssignedUserId == user.Id &&
```
Comment (jwinston-bn) [inline src/Aquifer.API/Endpoints/Resources/Content/MachineTranslation/Update/Endpoint.cs:22]: The problem with this is now anyone can update it. It's not a realistic scenario, but someone could go through and set a rating for everything in this status regardless of if they've worked on it or looked at it.

Comment (NateMerritt) [reply src/Aquifer.API/Endpoints/Resources/Content/MachineTranslation/Update/Endpoint.cs:22]: Do we need to lock it down to specific roles/permissions?

Comment (jwinston-bn) [reply src/Aquifer.API/Endpoints/Resources/Content/MachineTranslation/Update/Endpoint.cs:22]: Assigned editor is who sets it, so everyone has the permission.

Comment (jwinston-bn) [reply src/Aquifer.API/Endpoints/Resources/Content/MachineTranslation/Update/Endpoint.cs:22]: There's some sort of timing issue where the front-end reassigns before this gets set. I think the issue is on the front-end.

**Pull request closed without merging**